ui/src/importer/importerTaskDetails/ImportedDocuments.js

```diff
@@ -61,15 +99,107 @@ export class ImportedDocuments extends React.Component {
         });
       } else {
         this.setState({
-          data: response.data,
           isLoading: true,
+          data: response.data,
         });
       }
+      this.filteredRecords();
+      this.calculateMode();
+      this.calculateStatus();
+      this.calculateStatistics(response.data);
     } else {
       this.intervalId && clearInterval(this.intervalId);
     }
   };
 
+  calculateMode = () => {
+    const { data } = this.state;
+    switch (data.mode) {
+      case 'IMPORT':
+        this.mode = (
+          <Label color="blue" basic>
+            Import
+          </Label>
+        );
+        break;
+      case 'DELETE':
+        this.mode = (
+          <Label color="red" basic>
+            Delete
+          </Label>
+        );
+        break;
+      case 'PREVIEW_IMPORT':
+        this.mode = (
+          <Label color="teal" basic>
+            Preview (import)
+          </Label>
+        );
+        break;
+      case 'PREVIEW_DELETE':
+        this.mode = (
+          <Label color="teal" basic>
+            Preview (delete)
+          </Label>
+        );
+        break;
+      case 'ERROR':
+        this.mode = (
+          <Label color="red" basic>
+            Error
+          </Label>
+        );
+        break;
+      default:
+        throw Error('There was a problem processing the mode.');
+    }
+  };
+
+  statusLabel = () => {
+    const { data, isLoading } = this.state;
+    if (!data) return <Label color="grey">Fetching status</Label>;
+    if (isLoading)
+      return (
+        <Label color="blue">
+          <Icon loading name="circle notch" /> Importing literature
+        </Label>
+      );
+    switch (data.status) {
+      case 'SUCCEEDED':
+        return <Label color="green">Import successfull</Label>;
+      case 'CANCELLED':
+        return <Label color="yellow">Import cancelled</Label>;
+      default:
+        return <Label color="red">Import failed</Label>;
+    }
+  };
+
+  calculateStatus = () => {
+    this.status = this.statusLabel();
+  };
+
+  calculateStatistics = data => {
+    const { statistics } = this.state;
+    let stats = _cloneDeep(statistics);
+    const records = data.records;
+    const newRecords = records.slice(this.lastIndex, records.length);
+
+    this.lastIndex += newRecords.length;
+    stats.records.value = data.loaded_entries + '/' + data.entries_count;
```

Comment:
```suggestion
    stats.records.value = `${data.loaded_entries}/${data.entries_count}`;
```
addition on strings and numbers in javascript sometimes does not end well :)